### ISSUE TITLE
[Reading] Constitutional AI: Harmlessness from AI Feedback

### ISSUE THREAD

**Issue by hiroaki-1-1-0** [hiroaki-1-1-0/Reading-Papers#9]
## 0. Info
- Author: 
- Paper: [link](https://arxiv.org/abs/2212.08073)

## 1. In a Nutshell
- Prepare training data according to the principles without using manually labeled data, perform SFT, and then perform RLAIF for harmfulness.

## 2. Novelty / Utility
- Replaces human harmfulness labels with RLAIF (RL from AI feedback) and uses critique→revision to self-improve; chain-of-thought boosts label quality and transparency.

## 3. Core Method
- The process has two main stages:

    - Stage 1 (SL-CAI): Sample harmful replies, have the model critique them via constitutional rules, then revise and fine-tune on revisions.
 
    - Stage 2 (RL-CAI): Generate answer pairs, have a feedback model choose the safer one (multiple-choice under principles), train a preference model, then RL fine-tune against it. 

<img width="784" alt="Image" src="https://github.com/user-attachments/assets/c3bef8cb-9cba-429c-afa6-c42c88c66e3a" />

## 4. Validation
- Crowdworker Elo shows higher harmlessness at similar helpfulness vs RLHF baselines; models are notably less evasive.

<img width="662" height="1213" alt="Image" src="https://github.com/user-attachments/assets/59881549-7781-4a71-b05e-a8308fc6866a" />

## 5. Limitations / Discussions
- Principles are ad-hoc; still rely on human labels for helpfulness.
- Over-optimization can cause boilerplate, over-cautious tone.
- Dual-use potential, as it could be used to train malicious AI systems more easily.

## 6. Further Reading
- Bai et al. (2022), “Training a Helpful and Harmless Assistant with RLHF”: Baseline RLHF approach this work builds on—in a nutshell, human-preference-driven alignment.

## 7. Personal Comments
- Clean, practical recipe for scaling supervision
- one of the biggest open need is principled constitution design in long-run deployment. 
- RLHF is outdated for alignment, so I guess they might use DPO or RL for safety alignment in these days.
